src/rgw/driver/rados/rgw_sal_rados.cc

```diff
@@ -3018,77 +3099,105 @@ int RadosMultipartUpload::abort(const DoutPrefixProvider *dpp, CephContext *cct,
   int ret;
   uint64_t parts_accounted_size = 0;
 
-  do {
-    ret = list_parts(dpp, cct, 1000, marker, &marker, &truncated, y);
+  prefix_map_t processed_prefixes;
+
+  static constexpr auto MAX_DELETE_RETRIES = 15u;
+  for (auto i = 0u; i < MAX_DELETE_RETRIES; i++) {
+    ret = meta_obj->get_obj_attrs(y, dpp);
     if (ret < 0) {
-      ldpp_dout(dpp, 20) << __func__ << ": RadosMultipartUpload::list_parts returned " <<
-	ret << dendl;
-      return (ret == -ENOENT) ? -ERR_NO_SUCH_UPLOAD : ret;
+      ldpp_dout(dpp, 0) << __func__ << ": ERROR: failed to get obj attrs, obj=" << meta_obj
+                        << " ret=" << ret << dendl;
+      return ret;
```

Comment:
the call to `list_parts()` has a special case for `ERR_NO_SUCH_UPLOAD`. i guess we need the same for the new `meta_obj->get_obj_attrs()` call here?
```
2024-08-21T21:06:47.202 INFO:teuthology.orchestra.run.smithi016.stdout:=================================== FAILURES ===================================
2024-08-21T21:06:47.202 INFO:teuthology.orchestra.run.smithi016.stdout:____________________ test_abort_multipart_upload_not_found _____________________
2024-08-21T21:06:47.202 INFO:teuthology.orchestra.run.smithi016.stdout:
2024-08-21T21:06:47.202 INFO:teuthology.orchestra.run.smithi016.stdout:    def test_abort_multipart_upload_not_found():
2024-08-21T21:06:47.202 INFO:teuthology.orchestra.run.smithi016.stdout:        bucket_name = get_new_bucket()
2024-08-21T21:06:47.202 INFO:teuthology.orchestra.run.smithi016.stdout:        client = get_client()
2024-08-21T21:06:47.202 INFO:teuthology.orchestra.run.smithi016.stdout:        key="mymultipart"
2024-08-21T21:06:47.202 INFO:teuthology.orchestra.run.smithi016.stdout:        client.put_object(Bucket=bucket_name, Key=key)
2024-08-21T21:06:47.203 INFO:teuthology.orchestra.run.smithi016.stdout:
2024-08-21T21:06:47.203 INFO:teuthology.orchestra.run.smithi016.stdout:        e = assert_raises(ClientError, client.abort_multipart_upload, Bucket=bucket_name, Key=key, UploadId='56788')
2024-08-21T21:06:47.203 INFO:teuthology.orchestra.run.smithi016.stdout:        status, error_code = _get_status_and_error_code(e.response)
2024-08-21T21:06:47.203 INFO:teuthology.orchestra.run.smithi016.stdout:        assert status == 404
2024-08-21T21:06:47.203 INFO:teuthology.orchestra.run.smithi016.stdout:>       assert error_code == 'NoSuchUpload'
2024-08-21T21:06:47.203 INFO:teuthology.orchestra.run.smithi016.stdout:E       AssertionError: assert 'NoSuchKey' == 'NoSuchUpload'
2024-08-21T21:06:47.203 INFO:teuthology.orchestra.run.smithi016.stdout:E
2024-08-21T21:06:47.203 INFO:teuthology.orchestra.run.smithi016.stdout:E         - NoSuchUpload
2024-08-21T21:06:47.203 INFO:teuthology.orchestra.run.smithi016.stdout:E         + NoSuchKey
2024-08-21T21:06:47.203 INFO:teuthology.orchestra.run.smithi016.stdout:
2024-08-21T21:06:47.203 INFO:teuthology.orchestra.run.smithi016.stdout:s3tests_boto3/functional/test_s3.py:6302: AssertionError
```

Reply:
fixed